### PR TITLE
Fixed all existing INDEX statements and added missing ones

### PR DESCRIPTION
**Pull request by Hydrocharged**
I don't have write permissions on our `liquidata-inc/vitess` repository for some reason, so here we go! I'll close this when I get it but you can review it from here for now.

### CODE REVIEW

File: go/vt/sqlparser/sql_test.go [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+	res IndexAlteration
+}
+func testIndex(t *testing.T, tests []testIndexStruct) {
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("%v", test.statement), func(t *testing.T) {
+			res, err := Parse(test.statement)
+			require.NoError(t, err)
+			ddlRes, ok := res.(*DDL)
+			require.True(t, ok)
+			assert.Equal(t, AlterStr, ddlRes.Action)
```
Comment (zachmu) [inline go/vt/sqlparser/sql_test.go]: You should be able to just use assert.Equal on the expected and actual structs, no need to compare every field explicitly like this.

File: go/vt/sqlparser/sql.y [hunk 2]
```diff
@@ -1403,10 +1412,21 @@ alter_table_statement:
     // Change this to a rename statement
     $$ = &DDL{Action: RenameStr, FromTables: TableNames{$4}, ToTables: TableNames{$7}}
   }
-| ALTER ignore_opt TABLE table_name RENAME index_opt skip_to_end
+| ALTER ignore_opt TABLE table_name ADD index_opt sql_id '(' index_column_list ')' index_option_list_opt  skip_to_end
```
[the remark this answers is not in the document]
Comment (Hydrocharged) [reply go/vt/sqlparser/sql.y]: Looks like `index_opt` and `index_or_key` are the exact same rules. `index_opt` isn't for an optional index.

**Comment by Hydrocharged** [comment]
I made sure to resolve all shift/reduce errors before committing, so no need to worry about any of those.

File: go/vt/sqlparser/parse_test.go [hunk 3]
```diff
@@ -903,10 +903,10 @@ var (
 		output: "rename table a to b",
 	}, {
 		input:  "alter table a rename index foo to bar",
-		output: "alter table a",
+		output: "alter table a rename index foo to bar",
```
Comment (zachmu) [inline go/vt/sqlparser/parse_test.go]: For tests where input == output, you can omit output

File: go/vt/sqlparser/sql.y [hunk 4]
```diff
@@ -1281,6 +1305,29 @@ constraint_info:
     $$ = &ForeignKeyDefinition{Source: $4, ReferencedTable: $7, ReferencedColumns: $9, OnDelete: $11, OnUpdate: $12}
   }
 
+show_from:
```
Comment (zachmu) [inline go/vt/sqlparser/sql.y]: call this from_or_in

File: go/vt/sqlparser/sql.y [hunk 5]
```diff
@@ -1217,6 +1227,20 @@ index_info:
     $$ = &IndexInfo{Type: string($1), Name: NewColIdent($2), Unique: false}
   }
 
+indexes_or_keys:
+  INDEX
+  {
+    $$ = string($1)
```
Comment (zachmu) [inline go/vt/sqlparser/sql.y]: For rules like this where all matches evaluate to the same thing, you can just declare the evaluation for the first one and omit it for the others. Check out explainable_statement for an example.

**Pull request merged**